### PR TITLE
Update Http\Client docs for PSR7 changes.

### PR DESCRIPTION
**Pull request by markstory**
Make the docs have fewer API reference like by including more human friendly sectioning and examples with comments.

Refs cakephp/cakephp#6960

### CODE REVIEW

File: en/core-libraries/httpclient.rst [hunk 1]
```diff
@@ -428,6 +396,63 @@ XML data is decoded into a ``SimpleXMLElement`` tree::
...
+    // Get the response encoding
+    $response->getEncoding();
+
+    // Get an array of key=>value for all headers
+    $response->headers;
+
+Accessing Cookie Data
+---------------------
+
+You can read cookies with either a few different methods depending on how much
```
Comment (ravage84) [inline en/core-libraries/httpclient.rst]: "either" seems superfluous.

File: en/core-libraries/httpclient.rst [hunk 2]
```diff
@@ -428,6 +396,63 @@ XML data is decoded into a ``SimpleXMLElement`` tree::
 The decoded response data is stored in the response object, so accessing it
 multiple times has no additional cost.
 
+Accessing Response Headers
+--------------------------
+
+You can access headers through a few different methods. Header names are always
+treated as case-insensitive values when accessing them through methods::
+
+    // Get all the headers with array values.
```
Comment (ravage84) [inline en/core-libraries/httpclient.rst]: How about "Get all the headers as associative array"?

>      * @return string[][] Returns an associative array of the message's headers.
     *     Each key MUST be a header name, and each value MUST be an array of
     *     strings for that header.

http://www.php-fig.org/psr/psr-7/#3-3-psr-http-message-responseinterface

File: en/core-libraries/httpclient.rst [hunk 3]
```diff
@@ -428,6 +396,63 @@ XML data is decoded into a ``SimpleXMLElement`` tree::
...
+You can read cookies with either a few different methods depending on how much
+data you need about the cookies::
+
+    // Get all cookies (full data)
+    $response->getCookies();
+
+    // Get a single cookie's value.
+    $response->getCookie('session_id');
+
+    // Get a the complete data for a single cookie
```
Comment (ravage84) [inline en/core-libraries/httpclient.rst]: What's "complete" in that context?

**Comment by markstory** [comment]
Thanks for the feedback @ravage84 I've updated the docs.

**Pull request merged**